v2/spiffetls/tlsconfig/config_test.go

```diff
@@ -311,10 +418,20 @@ func TestGetClientCertificate(t *testing.T) {
 		},
 	}
 
+	localTrace := tlsconfig.Trace{
+		GetCertificate: func() interface{} {
+			fmt.Printf("got start of GetTLSCertificate\n")
+			return nil
+		},
+		GotCertificate: func(interface{}, tlsconfig.GotCertificateInfo) {
+			fmt.Printf("got end of GetTLSCertificate\n")
+		},
+	}
+
 	for _, testCase := range testCases {
 		testCase := testCase
 		t.Run(testCase.name, func(t *testing.T) {
-			getClientCertificate := tlsconfig.GetClientCertificate(testCase.source)
+			getClientCertificate := tlsconfig.GetClientCertificate(testCase.source, tlsconfig.WithTrace(localTrace))
```

Comment:
It would be good to assert that `GetCertificate` and `GotCertificate` executed as expected. We may check what was written to stdout.